pkg/registry/generic/etcd/etcd.go

```diff
@@ -403,6 +403,7 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 			if lastGraceful > 0 {
 				return out, nil
 			}
+			return out, nil
 			// fall through and delete immediately
```

Comment:
> This block is basically now just return out, nil

Actually rest.BeforeDelete() at line 387 may change the `*options.GracePeriodSeconds` to 0, so I kept this block.